.github/workflows/build.yml

```diff
@@ -1,6 +1,9 @@
 name: Build
 
-on: [push, pull_request]
+on:
+  pull_request:
+    types: [opened, reopened]
```

Comment:
My understanding is that `push` is only triggered when you push to the repository itself.  So if you push more commits to a pull request coming from a different repository, then the workflow won't run as you currently have it configured.

I think it is more important that we don't skip runs than that we avoid duplicate runs.

In general pull requests more commonly come from a different repository, and duplicate runs aren't that big of a deal anyway.

Maybe you could avoid duplicate runs by checking in the case of a pull_request event that it is coming from a different repository.  But then I guess it wouldn't show the results in the pull request itself?

Reply:
Its strange to me that my push events are only executing on my fork. Maybe because it is indeed a fork, but I use the same setup on other forks, and my push events trigger a run on upstream when a PR is open (from my fork).

There might be something up with your repo's settings in this regard.